app/actions/route_update.rb

```diff
@@ -2,12 +2,15 @@ module VCAP::CloudController
   class RouteUpdate
     def update(route:, message:)
       Route.db.transaction do
-        route.options = route.options.symbolize_keys.merge(message.options).compact if message.requested?(:options)
-
+        if message.requested?(:options)
+          route.options = route.options.symbolize_keys.merge(message.options).compact
+          route.apps.each do |process|
```

Comment:
I think it would be better to move this communication with Diego outside of the database transaction, so in pseudo code:
```
transaction (
  if new/changed options (
    update options
    save
  )
  update metadata
)

if new/changed options (
  notify diego
)

return route
```